libraries/AP_InertialSensor/AP_InertialSensor_Invensensev3.cpp

```diff
@@ -609,13 +675,47 @@ void AP_InertialSensor_Invensensev3::set_filter_and_scaling_icm42670(void)
     register_write(INV3REG_70_ACCEL_CONFIG1, 0x01);
 }
 
+/*
+  set the filter frequencies and scaling for the ICM-456xy
+ */
+void AP_InertialSensor_Invensensev3::set_filter_and_scaling_icm456xy(void)
+{
+    backend_rate_hz = 1600;
```

Comment:
This looks like it is pretty configurable. I really think we should try and support the logic in set_filter_and_scaling() to support multiples of 1.6Khz. I presume it also has a hardware AAF that would need configuring properly. OTOH maybe its worth getting something working in first before playing with these settings.